packages/swap/src/payment.ts

```diff
@@ -10,13 +10,17 @@ export class Payment extends Contract {
   _amount: number
   _owners: string[]
 
-  constructor(owner: string, _amount: number) {
-    super({ _owners: [owner], _amount })
+  constructor(_amount: number) {
+    super({ _amount })
   }
 
   transfer(to: string) {
     this._owners = [to]
   }
+
+  setAmount(a: number) {
+    this._amount = a
+  }
 }
 
 export class PaymentMock {
```

Comment:
It can be useful to implement from an interface, to warrantee that PaymentMock and Payment have the same prototype (if possible)